communication/nats/dialog/dialog_waiter.go

```diff
@@ -70,10 +74,21 @@ func (waiter *dialogWaiter) ServeDialogs(dialogHandler communication.DialogHandl
 		if request.PeerID == "" {
 			return &responseInvalidIdentity, nil
 		}
+		registered, err := waiter.identityRegistry.IsRegistered(common.HexToAddress(request.PeerID))
```

Comment:
Idea was to have `dialogWaiter` thin and all business specific logic goes to `DialogHandler.Handle(Dialog)`, which validates before handling

Reply:
Ok. let's see if I can improve that

Reply:
Moved to dialog handler. Well looks nicer